sample-skill/lambda/lambda_function.py

```diff
@@ -59,45 +52,101 @@ def handle(self, handler_input: HandlerInput) -> Response:
 MAX_TIMEOUT = 5
 
 
-def get_jwt_token() -> tuple:
-    """Return the jwt token to consume the API"""
-    error_message, access_token = None, None
+def get_jwt_token() -> str | None:
+    """Return the jwt token to consume the API
 
+    Returns:
+        str: JWT token
+        None: If the token can't be retrieved
+    """
     url = f"{ENDPOINT}/oauth2/access_token"
     payload = f'client_id={CLIENT_ID}&client_secret={CLIENT_SECRET}&grant_type={GRANT_TYPE}'
     headers = {'Content-Type': 'application/x-www-form-urlencoded'}
 
     response = requests.post(url, data=payload, headers=headers, timeout=MAX_TIMEOUT)
 
     if response.status_code == HTTPStatus.OK:
-        access_token = response.json()["access_token"]
-    else:
-        error_message =  "No fue posible consultar el progreso por un error de acceso."
+        return response.json()["access_token"]
 
-    return access_token, error_message
+    return None
 
 
-def get_course_progress(username: str, coursename: str) -> str:
-    """Return the progress of an user in a course"""
-    access_token, error_message = get_jwt_token()
+def get_course_progress(username: str, course_id: str, jwt_token: str) -> float | None:
+    """Return the progress of an user in a course
 
-    if not access_token:
-        return error_message
+    Args:
+        username (str): Username of the student
+        course_id (str): Course id
+        jwt_token (str): JWT token to consume the API
 
+    Returns:
+        float: Progress of the student in the course
+        None: If the progress can't be retrieved
+    """
     url = f"{ENDPOINT}/eox-core/api/v1/grade/"
-    payload = {"username": username, "course_id": COURSES.get(coursename)}
-    headers = headers = {"Authorization": f"Bearer {access_token}"}
+    payload = {"username": username, "course_id": course_id}
+    headers = headers = {"Authorization": f"Bearer {jwt_token}"}
```

Comment:
Have you tried this? this should not work because the authorization header must say `jwt`.

Also this PR should ADD the capacity to get the jwt token when required. Never to remove the Bearer access token.

The eox-core api only supports bearer token.
The enrollments API only supports jwt token.

We will have to use both.

Reply:
Hi @felipemontoya 

I was testing both APIs (eox-core, enrollments) and could consume the endpoints only with the Bearer token. Will it still be necessary to include the capacity to obtain a JWT token?

Reply:
I don't believe so. If bearer token work for both, let's leave it as is.